Supervision Work/src/uk/ac/cam/ap886/oopjava/supervisionProjects/Hangman/ArrayDictionary.java

```diff
@@ -0,0 +1,50 @@
+package uk.ac.cam.ap886.oopjava.supervisionProjects.Hangman;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Random;
+
+public class ArrayDictionary implements Dictionary {
+	private ArrayList <String> words=new ArrayList<String>();
+
+	//------------------------------------------
+	public ArrayDictionary(String filename){
+		try (BufferedReader r = new BufferedReader (new FileReader (filename));){
+			String temp = "";
+			do{
+				temp= r.readLine();
+				words.add(temp);
+			}while(temp!=null);
+		} catch (FileNotFoundException e) {
+			e.printStackTrace();	//Shouldn't happen
+		} catch (IOException e) {
+			e.printStackTrace();	//In case of Empty file
+		}
+		assert(!words.isEmpty());
```

Comment:
But I would argue that this is about who should be responsible for dealing with an empty dictionary, not about knowing early. If a dictionary is empty then as far as the dictionary is concerned, it has no reason to care. It's the game logic that cares. Either way, its a small point.